problems/data_attribute/problem.txt

```diff
@@ -0,0 +1,38 @@
+Let's assume common rails application view case:
+
+{{ code }}
+  # app/views/users/index.haml
+  <% @users.each do |u| %>
+    <p><%= u.name %> </p>
+  <% end %>
+{{/ code }}
+
+We iterate over collection and create some markup for each element. When adding some javascript to our view we may need to access data from collection than is not presented to the user.
+
+For example when we click on <p></p> element we want to sent ajax request with corresponding user id. Since we want user.name to be the only visible data from object we need to store id somewhere else.
+Common mistake here is to store user.id in class or id attribute in form of string key 'user-1'. We should avoid this because id and class attributes serves different purpose than
+storing logic data. For such operation we should use HTML data attribute. Syntax looks like this
+
+{{ code }}
+  <p data-id="1">...</p>
+{{/ code }}
+
+We can use data attributes as selectors to find element.
+
+{{ code }}
+  $('[data-name="exampleName"]');
```

Comment:
@pokorson I don't know if teaching usage of that kind selectors is a good thing. IMO they are slow and generally we should rather avoid them (maybe some tip here?)